### DISCUSSION TITLE
Single yaml file Github Actions

### DISCUSSION THREAD

**Discussion by MikhailTH** [community/community#26010] [Actions]
<p>Is there a way where you can use a single .yml file that was originally a workflow main.yml, that uses external actions, bash and node.js to create an action in itself without any other external dependencies? The goal was to create a single file tool that can be both run as an action or dumped into a repo and work as a workflow from there.</p>
<p>I’m fairly new to this and maybe missed something in documentation.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/mikhailth">@MikhailTH</a> ,</p>
<p>You can try to setup a <a href="https://docs.github.com/en/actions/creating-actions/about-actions#composite-run-steps-actions" rel="nofollow noopener">Composite run steps action</a>.<br>
The composite action is a type of actions, the other two types are <a href="https://docs.github.com/en/actions/creating-actions/about-actions#docker-container-actions" rel="nofollow noopener">Docker container action</a> and <a href="https://docs.github.com/en/actions/creating-actions/about-actions#javascript-actions" rel="nofollow noopener">JavaScript action</a>.</p>
<p>You can combine multiple workflow <em><strong>run</strong></em> steps within a composite action. However, it only can combine <em><strong>run</strong></em> steps, the <em><strong>uses</strong></em> steps to run other actions are not supported.</p>
<p>More details about creating composite action, you can see “<a href="https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action" rel="nofollow noopener"><strong>Creating a composite run steps action</strong></a>”.</p>

**Comment by MikhailTH** [comment]
<p>Yeah my action has both JS and a Bash shell running in it, I’m guessing that isn’t supported yet?</p>
<p>Edit:<br>
Any way around it other than having to call individual scripts from the repo?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/mikhailth">@MikhailTH</a> ,</p>
<p>Currently, we do have any available way that setup an action or reuse workflow to combine some <em><strong>run</strong></em> steps and <em><strong>uses</strong></em> steps.</p>
<ul>
<li>As mentioned above, the composite action can only combine <em><strong>run</strong></em> steps.</li>
<li>Unlike the YAML Pipelines of Azure Pipelines supports reusable <a href="https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&amp;tabs=example%2Cparameter-schema#template-references" rel="nofollow noopener"><strong>Templates</strong></a>, currently, GitHub does not support to reuse workflow.</li>
</ul>
<p>If you really need the similar features, I recommend that you can directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>
